types/src/on_chain_config/timed_features.rs

```diff
@@ -84,6 +82,7 @@ pub struct TimedFeaturesBuilder {
 }
 
 impl TimedFeaturesBuilder {
+    /// `timestamp` is a Unix Epoch timestamp in microseconds.
     pub fn new(chain_id: ChainId, timestamp: u64) -> Self {
```

Comment:
This is probably supper nit, but i think you can define a type alias `pub type Micros = u64`. Then all callers of `new(...)` have to case (and therefore think about units)?

another nit: `timestamp_micros`?